tests/generator/base.py

```diff
@@ -274,6 +274,7 @@ def validate_generated_yaml(self, yaml_input):
                     input_yaml = yaml.safe_load('')
 
             # Read output of the YAML generator (if any)
+            output_yaml = None
```

Comment:
This seems unrelated. It will be overridden in either the `if` or the `else` statement. It is best practice to have a variable clearly defined on top before we do any logic on it, so we can leave it in there.
But just to clarify: This has no other purpose than better readability/clarity of the code, right?

Reply:
Oh, no, even better than that: it has no purpose. It's an accidental leftover of the previous version of #251 with its unfortunate side effect on empty files. I'll get rid of it.